File: AoC/2015/00.py
"""
"""

from time import time

def TrackTime(func):
    def wrapper(*a, **k):
        start = time()
        func(*a, **k)
        print("  >> in %f ms" % ((time() - start)*1000))
    return wrapper

def ParseInput(filename):
    data = []
    with open(filename, 'r') as f:
        for l in f:
            data.append(l.strip())
    f.close()
    return data

@TrackTime
def Part1(data):
    total = 0
    print(f'Answer to part 1 is {total}')

@TrackTime
def Part2(data):
    total = 0
    print(f'Answer to part 1 is {total}')


data = ParseInput('./data/06.input')
Part1(data)
Part2(data)

File: AoC/2024/00.js
const DAY = 'XX';
const fs = require( 'fs');
let filename = `./data/${DAY}`;

const parseInput = () => {
    let data = fs.readFileSync(filename, 'utf8');
    return data;
}

const part1 = (data) => {
    console.time('Time');
    let res = 0;

    console.log('Part 1:', res);
    console.timeEnd('Time');
}

const part2 = (data) => {
    console.time('Time');
    let res = 0;

    console.log('Part 2:', res);
    console.timeEnd('Time');
}

if (process.argv.length > 2) {
    filename += '.test';
}
console.log('Advent Of Code 2024 day',  DAY);
console.log('==========================');
part1(parseInput());
console.log('--------------------------');
part2(parseInput());